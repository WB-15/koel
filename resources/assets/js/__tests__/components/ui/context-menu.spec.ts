import Component from '@/components/ui/ContextMenuBase.vue'
import { mount } from '@/__tests__/adapter'

declare const global: any

describe('components/ui/ContextMenuBase', () => {
  afterEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('renders', () => {
    expect(mount(Component)).toMatchSnapshot()
  })

  it('renders extra CSS classes', async () => {
    const wrapper = mount(Component, {
      propsData: {
        extraClass: 'foo'
      }
    })
    await (wrapper.vm as any).open(0, 0)
    expect(wrapper.find('.menu').hasClass('foo')).toBe(true)
  })

  it('opens', () => {
    const wrapper = mount(Component)
    ;(wrapper.vm as any).open(42, 128)
    expect(wrapper.find('.menu').element.style.top).toBe('42px')
    expect(wrapper.find('.menu').element.style.left).toBe('128px')
    expect(global.getComputedStyle(wrapper.find('.menu').element).display).toBe('block')
  })

  it('closes', async () => {
    const wrapper = mount(Component)
    await (wrapper.vm as any).open(42, 128)
    ;(wrapper.vm as any).close()
    expect(wrapper.html()).toBeUndefined()
  })
})
