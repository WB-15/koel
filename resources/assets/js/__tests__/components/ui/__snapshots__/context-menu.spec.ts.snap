// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/context-menu renders 1`] = ``;
